tests/testthat/test-replace_code_chunks.R

```diff
@@ -0,0 +1,8 @@
+test_that("filter_ends correctly filters end blocks", {
+  starts = c(FALSE, FALSE, FALSE, TRUE, FALSE, FALSE)
+  ends = c(TRUE, FALSE, TRUE, FALSE, FALSE, TRUE)
+  true_ends = c(FALSE, FALSE, FALSE, FALSE, FALSE, TRUE)
+  expect_equal(filter_ends(starts, ends), true_ends)
+  expect_equal(filter_ends(starts, true_ends), true_ends)
+  expect_equal(filter_ends(rep(starts, 2), rep(ends, 2)), rep(true_ends, 2))
+})
```

Comment:
Checking my understanding here - this test is appropriate because `filter_ends` relies on starts and ends essentially being paired?

Reply:
Correct. The problem is code blocks that are not executable chunks (just generic fenced code) will be picked up by the "end" regex, so  we need to get rid of the ones that don't come first after a "start" regex match, which is specific to executable chunks.